### PR TITLE
Fix cUSD faucet transfers

### LINKED ISSUE #133: Investigate why cUSD faucet transfers stopped
From @ericnakagawa on [Slack](https://celo-org.slack.com/archives/C014SRL4URY/p1702181327509849): 
> Hi Arthur and team, we’re investigating some issues reported at the ETH Global Hackathon. The [cUSD wallet](https://celoscan.io/address/0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1) distributions stopped 20 days back. CELO token was being distributed, but no mento Stables.

### PR DESCRIPTION
**Pull request by shazarre**
### Description

This PR fixes an issue where faucet was not sending any stable token transfers. It was failing because calling [this line](https://github.com/celo-org/developer-tooling/blob/f1677581b90675e37a4846ce53b29d8615a056e6/packages/sdk/contractkit/src/celo-tokens.ts#L199) throws an error and the function returns `false` resulting in returning no tokens at all.

### Other changes

None.

### Tested

Tested locally by running a custom script that invoked function transferring the tokens. Observed in the logs that transactions were sent out for those.

### Related issues

- Fixes #133 

### Backwards compatibility

Backwards compatible.

### CODE REVIEW

File: apps/firebase/src/celo-adapter.ts [hunk 1]
```diff
@@ -137,99 +137,76 @@ export class CeloAdapter {
...
 
-            const swapTxObj = await mento.swapOut(
-              celoToken.address,
-              stableTokenAddr,
-              realAmount.toString(),
-              maxCeloToTrade.toString(),
-            )
-            console.info('swap TX', swapTxObj)
-            await this.signer.sendTransaction(swapTxObj)
-          } else {
```
Comment (aaronmgdr) [inline apps/firebase/src/celo-adapter.ts:203]: probably makes sense to remove the related `if mento` tooo

**Review by aaronmgdr — APPROVED** [review]
bardzo dobrze!

**Comment by arthurgousset** [comment]
Successfully deployed to staging ✅

I'm looking for a way to test the staging faucet. 
Current idea is to use the Vercel preview, which seems to point to the staging faucet app (based on the env variables). 

Unfortunately, when I faucet from in the Vercel preview, the request fails because the recaptcha doesn't work

From the Vercel logs:

```
POST /api/faucet

Faucet Failed due to Recaptcha [ 'browser-error' ]
```

image

When I visit the preview, the captcha is indeed not working:

image

One idea is to authenticate with Github (top right), perhaps this bypasses the need for recaptcha. But Github authentication app doesn't work in Vercel preview, only in production.

**Comment by arthurgousset** [comment]
That means this PR fixes the cUSD bug in some cases, but not in others.

I hypothesise it has to do with the account balance of the requesting wallet. If it's too high, cStable requests don't work, if it's low enough cStable request work. 

We'll have to investigate this more.

**Comment by arthurgousset** [comment]
I propose we merge and deploy this PR as a partial fix, and submit a new PR if further changes are needed.
That way we unblock developers that wish to faucet cStables. At the moment, the faucet only makes CELO transfers and devs have to swap or mint at [app.mento.org](https://app.mento.org/).

File: apps/firebase/src/celo-adapter.ts (hunk 1 above)
Comment (arthurgousset) [reply apps/firebase/src/celo-adapter.ts:203]: > ## Deployment
> I propose we merge and deploy this PR as a partial fix.
> 
> That way we unblock developers that wish to faucet cStables. At the moment, the faucet only makes CELO transfers and devs have to swap or mint at [app.mento.org](https://app.mento.org/).
> 
> On that basis I'll merge this PR and deploy the firebase app to production.
> 
> ## Out-of-scope for this PR
> Then separately, make a new PR with:
> 
> * this [refactor proposed](https://github.com/celo-org/faucet/pull/156#discussion_r1473096120) by @aaronmgdr
> * any other changes that might be necessary to improve cStable disbursements.

Source: [comment in this PR](https://github.com/celo-org/faucet/pull/156#issuecomment-1924515888)

**Comment by arthurgousset** [comment]
## Deployed

✅ Successfully deployed from local machine to Firebase production with

```sh
$ yarn
$ yarn workspace @celo/faucet-app build
$ yarn workspace @celo/faucet-app build:rules
$ yarn workspace @celo/faucet-app deploy:prod
```

> [!NOTE]
> If you don't first run `$ yarn workspace @celo/faucet-app deploy:prod`, the firebase deployment will fail with
> 
> ```sh
> $ yarn workspace @celo/faucet-app deploy:prod                                                                    
>                                                                                                                                                     
> === Deploying to 'celo-faucet'...                                                                                                                   
>                                                                                                                                                     
> i  deploying database, functions                                                                                                                    
> Running command: yarn lint            
> # ...
> # ...
> Running command: yarn build                                                                                                                         
> Running command: yarn build:rules                                                                                                                   
> bolt: Generating database-rules.json...                                                                                                             
> ✔  functions: Finished running predeploy script.                                                                                                    
> i  database: checking rules syntax...                                                                                                               
>                                                                                                                                                     
> Error: Unexpected error while deploying database rules.  
> ````

**Comment by arthurgousset** [comment]
## Testing in production

✅ Worked as expected for a wallet with 0 CELO, see on [alfajores explorer](https://alfajores.celoscan.io/address/0x45c91983d5138a6836915A36293Bb2500035e96E#tokentxns).

image

As expected, didn't send cStables for a wallet with high balance.

**Pull request merged**